### PR TITLE
Cart managment

### PR DESCRIPTION
**Pull request by haarsh24**
1. Added Cart Management.
2. When clicked on Add to cart button, the product is added to the cart.
3. user can change the quantity of product.
4. price is calculated according to the product and its quantity.
5. After clicking on Add to cart button it changes to Go to cart and navigate the user to cart.
6. Currently badge is not working as expected, will fix it with wishlist PR.

### CODE REVIEW

**Review by xii-vi — COMMENTED** [review]
_Look good to merge.
I like how you break down components and store them differently._

File: src/components/checkoutCard/checkoutCard.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { useCart } from "../../contexts/cart-context";
+import { useEffect } from "react";
+
+const CheckoutCard = () => {
+  const {
+    cartState: { cartItem },
+  } = useCart();
+
+  const totalPrice = calculateTotalPrice(cartItem);
+  console.log(totalPrice);
```
Comment (rohit8686) [inline src/components/checkoutCard/checkoutCard.jsx:11]: better to remove console statements

File: src/components/navbar/navbar.jsx [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { Link } from 'react-router-dom';
+import { useCart } from "../../contexts/cart-context";
 import { useAuth } from "../../contexts/authContext";
 
 const Navbar = ()=>{
-    const { authState  } = useAuth();
+    const { authState } = useAuth();
+    const { cartState: { cartItem } } = useCart();
+    console.log(cartItem.length);
```
Comment (rohit8686) [inline src/components/navbar/navbar.jsx:10]: here also

**Review by rohit8686 — APPROVED** [review]
nice folder structure, looks good to merge

**Pull request merged**